package/ffnw-hoodselector/luasrc/hoodselector

```diff
@@ -0,0 +1,722 @@
+    if(wlan.quality ~= 0) then
+      table.insert(filtered_wlan_list, wlan)
+    end
+  end
+
+  return filtered_wlan_list
+end
+
+local function filter_default_hood_wlan_networks(default_hood, wlan_list)
+  local filtered_wlan_list = {}
+
+  for n,wlan in pairs(wlan_list) do
+    if(default_hood.bssid ~= wlan.bssid) then
+      table.insert(filtered_wlan_list, wlan)
+    end
+  end
+
+  return filtered_wlan_list
+end
+
+-- bool if direct VPN. The detection is realaise by searching the fastd network interface inside the originator table
+local function directVPN()
+  -- escape special chars "[]-"
+  for outgoingIF in io.open("/sys/kernel/debug/batman_adv/bat0/originators", 'r'):lines() do
+    local vpnIface = uci:get('fastd', 'mesh_vpn_backbone', 'net')
+    if not vpnIface then
+      io.stderr:write("fastd uci config broken! abort...\n")
+      exit()
+    end
+    if outgoingIF:match(string.gsub("%[  " .. vpnIface .. "%]","%_",'-'):gsub("%-", "%%-")) then
+      molwmtable["vpnrouter"] = "\"true\""
+      return true
+    end
+  end
+  molwmtable["vpnrouter"] = "\"false\""
+  return false
+end
+
+-- Retun a table of current peers from /etc/config/fastd
+local function getCurrentPeers()
+  local configPeers = {}
+  local err = uci:foreach('fastd', 'peer',
+    function(s)
+      if s['.name'] then
+        for prefix,peer in pairs(s) do
+          local tmpPeer = {}
+          if prefix:match(".name") then
+            if peer:match("mesh_vpn_backbone_peer_") then
+              -- val tmpRemote does not need uci exception check because its already include by "uci:foreach"
+              local tmpRemote = uci:get('fastd', peer, 'remote')
+              tmpRemote = tmpRemote[1]:split(" ")
+              local remote = {}
+              remote['host'] = tmpRemote[1]
+              remote[tmpRemote[2]] = tmpRemote[3]
+              -- uci:get does not need uci exception check because its already include by "uci:foreach"
+              tmpPeer['key'] = tostring(uci:get('fastd', peer, 'key'))
+              tmpPeer['remote'] = remote
+              configPeers[peer] = tmpPeer
+            end
+          end
+        end
+      end
+    end
+  )
+  if not err then
+    io.stderr:write("fastd uci config broken! abort...\n")
+    exit()
+  end
+  return configPeers
+end
+
+
+-- Get Geoposition. Return nil for no position
+local function getGeolocation()
+  local ret = {}
+  table.insert(ret, tonumber(uci:get('gluon-node-info', uci:get_first('gluon-node-info', 'location'), 'latitude')))
+  table.insert(ret, tonumber(uci:get('gluon-node-info', uci:get_first('gluon-node-info', 'location'), 'longitude')))
+  return ret
+end
+
+-- Return hood from the hood file based on geo position or nil, no real hood could be determined
+local function getHoodByGeo(jhood,geo)
+  for n, h in pairs(jhood) do
+    for n, box in pairs(h.boxes) do
+      if ( geo[1] >= box[1][1] and geo[1] < box[2][1] and geo[2] >= box[1][2] and geo[2] < box[2][2] ) then
```

Comment:
This only allows some kind of rectangular shape, which seems not to be suitable for practical use.